backend/compact-connect/stacks/api_stack/v1_api/query_providers.py

```diff
@@ -283,18 +310,46 @@ def _add_get_provider_ssn(
             threshold_metric_id='ad1',
         )
 
-        # We'll also set a flat maximum access rate of 5 reads per hour to alarm on
+        # We'll also set a flat maximum access rate of 5 reads per day to alarm on
+        # This will help us determine if the limit needs to be raised
         self.max_ssn_reads_alarm = Alarm(
             self.api,
             'MaxSSNReadsAlarm',
-            metric=metric,
+            metric=daily_read_ssn_count_metric,
             threshold=5,
             evaluation_periods=1,
             comparison_operator=ComparisonOperator.GREATER_THAN_THRESHOLD,
             treat_missing_data=TreatMissingData.NOT_BREACHING,
-            alarm_description=(f'{self.api.node.path} max ssn reads alarm'),
+            alarm_description=f'{self.api.node.path} max ssn reads alarm. The GET provider SSN endpoint has been '
+            f'invoked more than an expected threshold within a 24 hour period. Investigation is '
+            f'required to ensure access is not the result of abuse.',
         )
         self.max_ssn_reads_alarm.add_alarm_action(SnsAction(self.api.alarm_topic))
+        # Add an alarm for 429 responses from the SSN endpoint
+        self.ssn_api_throttling_alarm = Alarm(
```

Comment:
Because this metric is generated for _any_ 4XX response, having it alarm on any metrics at all is likely to have a lot of false-positives, since any random unauthenticated bot call to this method will set off an alarm. We should probably dial this threshold up enough to filter out the random crawler traffic. We already alarm on our own internal throttling, so this would be a backup to detect authn/authz failures that would otherwise be missed. I think any legitimate attack vector caught by this alarm would need to have a pretty high call volume to have a chance at success, so maybe we bump it up to like 100 in 5 mins?